aws/resource_aws_ssm_resource_data_sync.go

```diff
@@ -0,0 +1,180 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSsmResourceDataSync() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSsmResourceDataSyncCreate,
+		Read:   resourceAwsSsmResourceDataSyncRead,
+		Delete: resourceAwsSsmResourceDataSyncDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"bucket": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"region": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", strings.ToLower(m["bucket"].(string))))
+					buf.WriteString(fmt.Sprintf("%s-", strings.ToLower(m["region"].(string))))
+					if val, ok := m["key"]; ok {
+						buf.WriteString(fmt.Sprintf("%s-", strings.ToLower(val.(string))))
+					}
+					if val, ok := m["prefix"]; ok {
+						buf.WriteString(fmt.Sprintf("%s-", strings.ToLower(val.(string))))
+					}
+					return hashcode.String(buf.String())
```

Comment:
Can you please explain why do we need to lowercase all the values here? It seems unnecessary to me. 🤔